cirq/sim/wave_function_test.py

```diff
@@ -25,6 +25,102 @@ def assert_dirac_notation(vec, expected, decimals=2):
     assert cirq.dirac_notation(np.array(vec), decimals=decimals) == expected
 
 
+def test_bloch_vector_simple():
+    sqrt = np.sqrt(0.5)
+    test_state = np.array([sqrt, sqrt])
+
+    bloch = cirq.bloch_vector_from_state_vector(test_state, 0)
+    desired_simple = np.array([1,0,0])
+    np.testing.assert_array_almost_equal(bloch, desired_simple)
+
+
+def test_bloch_vector_invalid():
+    with pytest.raises(ValueError):
+        _ = cirq.bloch_vector_from_state_vector(
+            np.array([0.5, 0.5, 0.5]), 0)
+    with pytest.raises(IndexError):
+        _ = cirq.bloch_vector_from_state_vector(
+            np.array([0.5, 0.5,0.5,0.5]), -1)
+    with pytest.raises(IndexError):
+        _ = cirq.bloch_vector_from_state_vector(
+            np.array([0.5, 0.5,0.5,0.5]), 2)
+
+
+def test_bloch_vector_multi_pure():
+    test_state = np.array([0.5,0.5,0.5,0.5])
+
+    bloch_0 = cirq.bloch_vector_from_state_vector(test_state, 0)
+    bloch_1 = cirq.bloch_vector_from_state_vector(test_state, 1)
+    desired_simple = np.array([1,0,0])
+
+    np.testing.assert_array_almost_equal(bloch_1, desired_simple)
+    np.testing.assert_array_almost_equal(bloch_0, desired_simple)
+
+
+def test_bloch_vector_multi_mixed():
+    sqrt = np.sqrt(0.5)
+    test_state = np.array([sqrt, 0., 0., sqrt])
+
+    bloch_0 = cirq.bloch_vector_from_state_vector(test_state, 0)
+    bloch_1 = cirq.bloch_vector_from_state_vector(test_state, 1)
+    zero = np.zeros(3)
```

Comment:
Do a test involving five qubits, just to make sure the not-start-not-end case works.